### PR TITLE
Commerce 2392

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test

**Comment by andrea-ale-sbarra** [comment]
Hi @ethanbustad I'm getting this error, even if I didn't work on that module, is there something that I'm missing?

Thank you!

**Comment by ethanbustad** [comment]
ci:test
yeah that is weird. Let's try it again and see if that helps. I don't see how this could be an actual failure caused by your PR

File: commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v4_4_0/CommerceAddressUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+/**
+ * @author Andrea Sbarra
+ */
+public class CommerceAddressUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
```
Comment (marco-leo) [inline commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v4_4_0/CommerceAddressUpgradeProcess.java]: Extend BaseCommerceServiceUpgradeProcess and reuse addColumn method

File: headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountAddressResourceImpl.java [hunk 2]
```diff
@@ -53,6 +59,46 @@
...
+
+		if (commerceAddress == null) {
+			throw new NoSuchAddressException(
+				"Unable to find AccountAddress with externalReferenceCode: " +
+					externalReferenceCode);
+		}
+
+		_commerceAddressService.deleteCommerceAddress(
+			commerceAddress.getCommerceAddressId());
+		Response.ResponseBuilder responseBuilder = Response.noContent();
```
Comment (marco-leo) [inline headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountAddressResourceImpl.java:80]: missing empty line

File: headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountAddressResourceImpl.java [hunk 3]
```diff
@@ -97,7 +190,43 @@ public AccountAddress postAccountByExternalReferenceCodeAccountAddress(
...
+		}
+		else if (accountAddress.getExternalReferenceCode() != null) {
+			commerceAddress =
+				_commerceAddressService.fetchByExternalReferenceCode(
+					contextCompany.getCompanyId(),
+					accountAddress.getExternalReferenceCode());
+		}
+
+		if (commerceAddress != null) {
+			accountAddress = _toAccountAddress(
```
Comment (marco-leo) [inline headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountAddressResourceImpl.java]: return at this point

File: headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountAddressResourceImpl.java [hunk 4]
```diff
@@ -97,7 +190,43 @@ public AccountAddress postAccountByExternalReferenceCodeAccountAddress(
...
+					GetterUtil.getString(accountAddress.getCity(), null),
+					GetterUtil.getString(accountAddress.getZip(), null),
+					commerceAddress.getCommerceRegionId(),
+					commerceAddress.getCommerceCountryId(),
+					GetterUtil.getString(accountAddress.getPhoneNumber(), null),
+					GetterUtil.getInteger(
+						accountAddress.getType(), commerceAddress.getType()),
+					_serviceContextHelper.getServiceContext()));
+		}
+		else {
```
Comment (marco-leo) [inline headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountAddressResourceImpl.java]: no need of else

File: headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountAddressResourceImpl.java [hunk 5]
```diff
@@ -97,7 +190,43 @@ public AccountAddress postAccountByExternalReferenceCodeAccountAddress(
...
+			accountAddress = _addAccountAddress(
```
Comment (marco-leo) [inline headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountAddressResourceImpl.java]: return directly

**Review by marco-leo — CHANGES_REQUESTED** [review]

**Comment by andrea-ale-sbarra** [comment]
ci:test

**Comment by andrea-ale-sbarra** [comment]
ci:test

**Comment by andrea-ale-sbarra** [comment]
ci:test

**Pull request merged**